pkg/rbd/errors.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2019 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+// ErrImageNotFound is returned when image name is not found in the cluster on the given pool
+type ErrImageNotFound struct {
+	imageName string
+	err       error
+}
+
+func (e ErrImageNotFound) Error() string {
```

Comment:
Hmm.. this does not look like a common pattern for describing errors and method on top.

Reply:
Can you point to a common pattern to adopt? I do not know of any hence asking.

Reply:
@ShyamsundarR can we move this file outside of rbd, s that we can use same for cephfs also.
is this possible?

Reply:
Some errors move out to util when they are shared by CephFS (in the [refactor]https://github.com/ShyamsundarR/ceph-csi/commit/c987e8bc7c8999e1072db1a902e64efb256abae1) patch) some stay (e.g ErrImageNotFound) as they are RBD specific. So in this PR I cannot move it to a common file.

BTW, there is another common [file](https://github.com/ceph/ceph-csi/pull/312/files#diff-14a2aec463bbb3baabb04222dabe7fc7) in util with other errors already.

Reply:
@Madhu-1 pointed out to this source for error definition https://golang.org/src/errors/errors.go and as per this, the way errors are defined looks right. @humblec can you clarify if this looks right?